tests/fixtures/FinalOrAbstractClassLinter/classes.php.in

```diff
@@ -0,0 +1,17 @@
+<?hh
+
+class test {
+
+}
+
+final class test3 {
+
+}
+
+abstract class test4 {
+
+}
+
+interface interface1 {
+
+}
```

Comment:
newline at EOF

Reply:
also:
- add an abstract final class
- add XHP classes
- add traits

Reply:
Thanks for the feedback above! I've fixed them just now.

I wonder if we need to handle the `abstract final` class for this linter. That feels like a syntax error?

Reply:
`abstract final` is allowed in Hack; this is useful for classes that only have static methods (especially if you want to reference them via `classname<SomeSuperclass>`)

Reply:
Gotcha, the only reference I can find about this is a rejected PHP proposal for `abstract final` classes. So I was not sure if the same applies to Hack. Good to know!